src/api.ts

```diff
@@ -141,6 +142,41 @@ function createApi({
     }
   }
 
+  function getUpstreamTileUrl({
+    tilesetId,
+    zoom,
+    x,
+    y,
+  }: {
+    tilesetId: string
+    zoom: number
+    x: number
+    y: number
+  }) {
+    const tilesetRow:
+      | { tilejson: string; upstreamTileUrls?: string }
+      | undefined = db
+      .prepare('SELECT tilejson, upstreamTileUrls FROM Tileset where id = ?')
+      .get(tilesetId)
+
+    if (!tilesetRow) {
+      // TODO: Is this the appropriate error?
+      throw new NotFoundError(`Tileset id = ${tilesetId}`)
+    }
+
+    const tilejson: TileJSON = JSON.parse(tilesetRow.tilejson)
+
+    return tilesetRow.upstreamTileUrls
+      ? getInterpolatedUpstreamTileUrl({
+          tiles: JSON.parse(tilesetRow.upstreamTileUrls) as TileJSON['tiles'],
```

Comment:
should we wrap all json parses in try/catch? should we do the same for stringify too?

Reply:
Yeah, I think it's good practice to do, because it is something that can throw if somehow bad data gets into the DB. The stringify functions do not need try-catch because they do not throw errors (as far as I'm aware! - anyway typescript would catch that statically)